akka-stream-tests/src/test/scala/akka/stream/scaladsl/HubSpec.scala

```diff
@@ -199,28 +201,36 @@ class HubSpec extends StreamSpec {
       downstream.expectComplete()
     }
 
-    // Excluded in GH Actions: https://github.com/akka/akka/issues/30464
-    "immediately cancel new producers while draining" taggedAs GHExcludeTest in assertAllStagesStopped {
+    "immediately cancel new producers while draining" in assertAllStagesStopped {
       val downstream = TestSubscriber.probe[Int]()
       val (sink, draining) =
         MergeHub.sourceWithDraining[Int](16).take(20).toMat(Sink.fromSubscriber(downstream))(Keep.left).run()
       Source(1 to 10).runWith(sink)
       Source(11 to 20).runWith(sink)
-      draining.drainAndComplete()
 
-      downstream.request(10)
-      val firstBatch = downstream.expectNextN(10)
+      // Give the sources time to fill up the producer buffers before completing (and cancelling the sources)
```

Comment:
Do you think is possible to wait for a deterministic event/condition?

Reply:
I don't see a way

Reply:
what about having an `awaitAssert` or `eventually` on the first condition to be met?

Reply:
On what condition?

The problem we were seeing was (at least I strongly suspect ;) ) that `draining.drainAndComplete()` cancelled the sources before the sourced had had time to fill the buffers of the MergeHub. I don't see what we can wait for or retry to avoid that.